pandas/core/internals/construction.py

```diff
@@ -342,37 +338,29 @@ def ndarray_to_mgr(
 
         return ArrayManager(arrays, [index, columns], verify_integrity=False)
 
-    values = values.T
+    array = values.T
 
     # if we don't have a dtype specified, then try to convert objects
     # on the entire block; this is to convert if we have datetimelike's
     # embedded in an object type
-    if dtype is None and is_object_dtype(values.dtype):
-
-        if values.ndim == 2 and values.shape[0] != 1:
+    if dtype is None and is_object_dtype(array.dtype):
+        if array.ndim == 2 and array.shape[0] != 1:
             # transpose and separate blocks
-
-            dtlike_vals = [maybe_infer_to_datetimelike(row) for row in values]
-            dvals_list = [ensure_block_shape(dval, 2) for dval in dtlike_vals]
-
-            # TODO: What about re-joining object columns?
-            block_values = [
-                new_block(dvals_list[n], placement=n, ndim=2)
-                for n in range(len(dvals_list))
+            maybe_datetime = [
+                maybe_infer_to_datetimelike(instance) for instance in array
             ]
-
+            # don't convert (and copy) the objects if no type inference occurs
+            if any(
+                not is_dtype_equal(instance.dtype, array.dtype)
+                for instance in maybe_datetime
+            ):
+                return create_block_manager_from_arrays(
+                    maybe_datetime, columns, [columns, index]
+                )
         else:
-            datelike_vals = maybe_infer_to_datetimelike(values)
-            nb = new_block(datelike_vals, placement=slice(len(columns)), ndim=2)
-            block_values = [nb]
-    else:
-        nb = new_block(values, placement=slice(len(columns)), ndim=2)
-        block_values = [nb]
-
-    if len(columns) == 0:
-        block_values = []
+            array = maybe_infer_to_datetimelike(array)
 
-    return create_block_manager_from_blocks(block_values, [columns, index])
+    return create_block_manager_from_array(array, [columns, index])
```

Comment:
this change seems unnecessary.  id rather have this function return only here and not on L357